cmd/cli/plugin/feature/main.go

```diff
@@ -35,7 +36,13 @@ func main() {
 		FeatureDeactivateCmd,
 	)
 
+	ctx := signals.SetupSignalHandler()
```

Comment:
I'm not sure what sorts of plugins are being built, but certainly, some don't need context at all, or if a context is being passed, it's just a background context that doesn't do anything.

For example, a plugin can operate so quickly that there isn't even enough time for an interrupt or terminate signal to be sent before the program exits.

Even, as you say, "It's highly likely most plugins will be using some sort of context," should we make the default based on our opinion of what signal handler to use? If a plugin developer doesn't like our signals library (and, say, for example want to use the Kubernetes controller-runtime signals library) or doesn't even want to pass around a functioning context, should we make it so they have to update and change what we provided as default?

Based on your feedback in that if you think a default is the right way to go, even if it may have to be removed or changed depending on the plugin developer, I'll go ahead and add the line you highlighted, `ctx := signals.SetupHandler()` to the builder template.

Reply:
Let's keep it as is for now and evaluate as we see how plugins use context. But a couple of thoughts:

1. I can't think of a plugin that doesn't need to use context. Most, if not all, reach out to k8s API. A background context isn't good practice. So it's okay to mandate graceful shutdown.
2. We can be opinionated on an out-of-the-box signal handler that satisfies most use cases. If authors want to change it, they can as they have full control once the plugin has been scaffolded.